docs/networking/best-practice-vlan-aware-switch.md

```diff
@@ -0,0 +1,70 @@
+---
+sidebar_position: 50
+keywords:
+  - Harvester
+  - harvester
+  - Rancher
+  - rancher
+Description: Harvester is built on Kubernetes, which uses CNI as an interface between network providers and Kubernetes pod networking. Naturally, we implement the Harvester network based on CNI. Moreover, the Harvester UI integrates the Harvester network to provide a user-friendly way to configure networks for VMs.
+---
+
+# Best Practice for VLAN-aware Switch
+
+In this best practice guide for VLAN-aware switch, we will introduce Harvester VLAN network and external VLAN-aware switch configuration for common scenario.
+
+## Architecture
+
+Hardware:
+- 3 Harvester servers with only single port network card
+- VLAN-aware switch (taking Cisco-like configuration as example)
+
+Network Specification:
+- Management Network to control Harvester servers in VLAN 100
+- VM Network in VLAN 101-200
+
+Cabling:
+- Harvester servers connect to switch port from port 1 to 3
+
+The below diagram illustrates cabling in this guide.
+
+   ![vlan-aware-case.png](assets/vlan-aware-case.png)
+
+## External Switch Configuration
+
+In external switch, we takes Cisco-like configuration as example. Users should apply the following configurations to their switch.
+
+```
+switch# config terminal
+switch(config)# interface ethernet1/<Port Number>
+switch(config-if)# switchport
+switch(config-if)# switchport mode trunk
+switch(config-if)# switchport trunk allowed vlan 100-200
+switch(config-if)# switchport trunk native vlan 100
+switch(config-if)# no shutdown
+switch(config-if)# end
+switch# copy running-config startup-config
+```
+
+## Create a VLAN Network in Harvester
+
+A new VLAN network can be created via the **Advanced > Networks** page and clicking the **Create** button.
+
+Specify the name and VLAN ID that you want to create for the VLAN network <small>(You can specify the same vlan ID on different namespaces of [Rancher multi-tenancy](/rancher/virtualization-management/#multi-tenancy) support)</small>.
+   ![create-vlan-network.png](assets/create-network.png)
+
+### Connect VM to the same network of Harvester management network
+
+Once users finished the configuration in the previous section, external switch will send out untagged network traffic for manangement. Inside Harvester, the default VLAN tag to receive untagged taffic is VLAN 1.
+
+If users need VM connects to VLAN 100, management network, users can create a VLAN Network in Harvester with VLAN ID 1 configuration instead of VLAN ID 100.
```

Comment:
VM only connect to VLAN 1.

we got some issue that asked about how to let VM access same VLAN of `harvester-mgmt` if harvester-mgmt is under VLAN 100 for example.

harvester VLAN 1 and external switch VLAN 100 are same subnet (same L2) in example architecture.
harvester will add VLAN 1 tag to untagged packet while recv untagged traffic. (outside to VM VLAN1)
harvester will remove VLAN 1 tag from VLAN 1 traffic while sending out VLAN 1 tag traffic  (VM VLAN1 to outside)

external switch will add VLAN 100 tag to untagged packet while recv untagged traffic (harvester to external switch)
external switch will remove VLAN 100 tag from any VLAN 100 traffic while sending out to harvester (external switch to harvester)

So if harvester-mgmt is under VLAN 100 of external switch.
harvester-mgmt didn't need to deal with VLAN tag because external switch already remove it.
but if VM wants to connect to the same VLAN of harvester-mgmt, harvester-br0 will add VLAN 1 tag to all untagged packet.
So VM need to connect to VLAN 1 in harvester as well to have same L2 domain with harvester-mgmt and VLAN 100 in external switch

Reply:
I hope i didn't confuse you about this.
mixed untagged and tagged VLAN confuse people a lot.
so i put another figure for explaination, hope this will help
![image](https://user-images.githubusercontent.com/2821179/153384279-7fb725b8-3aa5-4970-b00f-4d1bf448e772.png)